src/com/dmdirc/logger/ErrorManager.java

```diff
@@ -298,10 +300,8 @@ public void sendError(final ProgramError error) {
     public void deleteError(final ProgramError error) {
         synchronized (errors) {
             errors.remove(error);
+            eventBus.publish(new ProgramErrorDeletedEvent(error));
```

Comment:
Don't really like firing events when holding locks on things... If anything that receives a ProgramErrorDeletedEvent tries to access the ErrorManager (which is not altogether unreasonable!) you'll end up with a nasty deadlock.

Could we just make errors a ConcurrentSkipListSet and drop all the synchronization?